gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -1059,12 +1094,20 @@ public void onStopContainerError(ContainerId containerId, Throwable t) {
     }
   }
 
-  //A class encapsulates Container instances, Helix participant IDs of the containers and Helix Tag
-  @AllArgsConstructor
+  // Class encapsulates Container instances, Helix participant IDs of the containers, Helix Tag, and
+  // initial startup command
   @Getter
-  static class ContainerInfo {
+  class ContainerInfo {
     private final Container container;
     private final String helixParticipantId;
     private final String helixTag;
+    private final String startupCommand;
+
+    public ContainerInfo(Container container, String helixParticipantId, String helixTag) {
+      this.container = container;
+      this.helixParticipantId = helixParticipantId;
+      this.helixTag = helixTag;
+      this.startupCommand = YarnService.this.buildContainerCommand(container, helixParticipantId, helixTag);
```

Comment:
Key part is that we store the start up command in the container map. And we can later print it in an error scenario